miner/miner.go

```diff
@@ -2,29 +2,98 @@ package miner
 
 import (
 	"math"
+	"sync"
 	"time"
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/ubclaunchpad/cumulus/blockchain"
 	"github.com/ubclaunchpad/cumulus/consensus"
 )
 
+// CurrentlyMining is a flag to control the miner.
+var CurrentlyMining bool
```

Comment:
Good call. Originally I wanted to have other packages check on them, but the exported function does that now.